sinks/influxdb/driver.go

```diff
@@ -85,101 +94,83 @@ func (self *influxdbSink) metricToSeries(timeseries *sink_api.Timeseries) *influ
 	if timeseries.MetricDescriptor.Type.String() != "" {
 		seriesName = fmt.Sprintf("%s_%s", seriesName, timeseries.MetricDescriptor.Type.String())
 	}
-
-	// Add the real metric value.
-	columns = append(columns, "value")
-	values = append(values, timeseries.Point.Value)
+	point := influxdb.Point{
+		Measurement: seriesName,
+		Tags:        make(map[string]string, 0, len(timeseries.Point.Labels)),
+		Fields: map[string]interface{}{
+			"value": timeseries.Point.Value,
+		},
+		Time: timeseries.Point.End.UTC(),
+	}
 	// Append labels.
-	if !self.c.avoidColumns {
-		for key, value := range timeseries.Point.Labels {
-			columns = append(columns, key)
-			values = append(values, value)
+	for key, value := range timeseries.Point.Labels {
+		if value != "" {
+			point.Tags[key] = value
 		}
-	} else {
-		seriesName = strings.Replace(seriesName, "/", "_", -1)
-		seriesName = fmt.Sprintf("%s_%s", util.LabelsToString(timeseries.Point.Labels, "_"), seriesName)
 	}
-	// Add timestamp.
-	columns = append(columns, "time")
-	values = append(values, timeseries.Point.End.Unix())
-	// Ass sequence number
-	columns = append(columns, "sequence_number")
-	values = append(values, self.seqNum.Get(seriesName))
 
-	return self.newSeries(seriesName, columns, values)
-}
-
-var eventColumns = []string{
-	"time",                     // Column 0
-	"sequence_number",          // Column 1
-	sink_api.LabelPodId.Key,    // Column 2
-	sink_api.LabelPodName.Key,  // Column 3
-	sink_api.LabelHostname.Key, // Column 4
-	"value",                    // Column 5
+	return point
 }
 
 // Stores events into the backend.
 func (sink *influxdbSink) StoreEvents(events []kube_api.Event) error {
-	dataPoints := []*influxdb.Series{}
+	if err := sink.createDatabase(); err != nil {
+		return err
+	}
 	if events == nil || len(events) <= 0 {
 		return nil
 	}
-	if !sink.c.avoidColumns {
-		dataPoint, err := sink.storeEventsColumns(events)
-		if err != nil {
-			glog.Errorf("failed to parse events: %v", err)
-			return err
-		}
-		dataPoints = append(dataPoints, dataPoint)
-	} else {
-		for _, event := range events {
-			dataPoint, err := sink.storeEventNoColumns(event)
-			if err != nil {
-				glog.Errorf("failed to parse events: %v", err)
-				return err
-			}
-			dataPoints = append(dataPoints, dataPoint)
-		}
-	}
-	err := sink.client.WriteSeriesWithTimePrecision(dataPoints, influxdb.Millisecond)
+	points, err := sink.eventsToPoints(events)
 	if err != nil {
+		glog.Errorf("failed to parse events: %v", err)
+		return err
+	}
+	bp := influxdb.BatchPoints{
+		Points:   points,
+		Database: sink.c.dbName,
+	}
+	if _, err = sink.client.Write(bp); err != nil {
+		if strings.Contains(err.Error(), dbNotFoundError) {
+			sink.dbExists = false
+		}
 		glog.Errorf("failed to write events to influxDB - %s", err)
 		sink.recordWriteFailure()
 	} else {
-		glog.V(1).Info("Successfully flushed events to influxDB")
+		glog.V(4).Info("Successfully flushed events to influxDB")
 	}
-	return err
 
+	return err
```

Comment:
It would be more idiomatic go to do `return err` after `recordWriteFailure()`, drop the `else` and return `nil` here.